### ISSUE TITLE
Documenso cannot be started on NixOS

### ISSUE THREAD

**Issue by alxpettit** [NixOS/nixpkgs#284354]
### Steps To Reproduce
1. compute the out path of Documenso (e.g., via `nix repl`)
2. `cd` to that path.
3. Try to run `npm run start`

### Expected behavior

Documenso should work locally on port 3000.

### Additional context

It looks like it's working first, but when you try to interact with it, it spits out [this error](https://gist.github.com/alxpettit/ac5f6b0fd70a5f863403adb38cae3e4f).

Key log segment:

```
PrismaClientInitializationError: Query engine library for current platform "linux-nixos" could not be found.
You incorrectly pinned it to linux-nixos
```

### Notify maintainers

@happysalada

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.8, NixOS, 23.11 (Tapir), 23.11.20231221.0b3d618`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/etc/nixPath`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by happysalada** [comment]
hey, thanks a lot for reporting this.
Unfortunately that doesn't ring a bell for me.
Maybe we could try to open an issue upstream about building from source.
Also one of the more recent versions might have fixed this already.
This might not be a lot of help, but if you find out a fix for this, I'd be happy to help you get it merged.

**Comment by alxpettit** [comment]
I'd love to help with this, although I'm not very familiar with the NPM ecosystem, and any issues I open may be poorly framed as a consequence. Upstream's documentation on 'development running' is pretty similar to our use case of grabbing the files, selecting the important ones, and eventually calling `nix run` -- so maybe that's a good starting point? The README.md instructs you to copy the `.env.example` file to `.env`, and possibly edit it with whatever configuration you want. I was wondering, why does that step seem to not show up in the [default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/documenso/default.nix)?

(Apologies if this is a red herring. As I said, I'm pretty unfamiliar with the NPM ecosystem.)

I'm also not sure what the difference is between `npm start dx` and `npm start run`, but I assume it's some sort of build/run script target similar to make?

I'm going to go ahead and try to prepare it standalone environment, and see what happens.

**Comment by happysalada** [comment]
hey, thank you for your work on this!
Just to give you some context on nixpkgs.
It's common to not set environment variables on the binary itself since those are a setting.
What usually happens is that the binary should be built properly and then a service is created where the proper environment variables are set to run it correctly. The binary itself should be independent of any settings so that anyone can later set them as they please.
if you want an example of a service where this is exactly what is done (just setting the environment variables to run a node js service), you can grep the following line
```
      systemd.services.lemmy-ui = {
```
lemmy-ui is an example of just that.

**Comment by alxpettit** [comment]
Yeah, setting it in the package, since it's immutable, would usually mean rebuilding the package each time it changes I imagine. Although, for now, I've just been trying to figure out how to get it working in the first place. I am not exactly sure at what point the NPM bootstrapping process injects the environment in, or how -- and I don't even know if it's literal environment variables or something else. I was just playing around to see if I could start with a working state, and then have something to compare to my broken state to hint as to what's going wrong.

(It's also notable that there are some oddities -- like, if I recall properly, the Wordpress package does bake the config.php in with the package. I was actually considering making a PR to have that be a symlink to a data directory, which is then optionally either separately managed by a different, smaller package -- or managed imperatively if the user likes to roll that way.)

**Comment by happysalada** [comment]
completely understand, thank you for your efforts in trying to fix this!
